atlasdb-config/src/main/java/com/palantir/atlasdb/factory/LockServices.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.palantir.atlasdb.config.ServerListConfig;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.conjure.java.api.config.service.ServicesConfigBlock;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.dialogue.clients.DialogueClients;
+import com.palantir.lock.LockRpcClient;
+import com.palantir.lock.LockService;
+import com.palantir.lock.client.LockRefreshingLockService;
+import com.palantir.lock.client.RemoteLockServiceAdapter;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.concurrent.ExecutorService;
+
+public final class LockServices {
+    // No need for services config block since timelock server uris come from runtime config.
+    private static final Refreshable<ServicesConfigBlock> DUMMY_SERVICES_CONFIG =
+            Refreshable.only(ServicesConfigBlock.builder().build());
+
+    private LockServices() {
+        // static factory
+    }
+
+    public static LockService createLockServiceClient(
+            Refreshable<ServerListConfig> timeLockServerListConfig,
+            ExecutorService lockClientExecutor,
+            UserAgent userAgent,
+            MetricRegistry metricRegistry,
+            TaggedMetricRegistry taggedMetricRegistry,
```

Comment:
I guess a TaggedMetricRegistry is always a superset of the MetricRegistry. But there is no way to avoid receiving both metric registries, since we can't guarantee that they're composed from the same object right?

Reply:
They're actually different "parallel" classes: `MetricRegistry` is the "generic" coda-hale one, and `TaggedMetricRegistry` is a Tritium thing that is kind of a parallel.

We could switch everyone to use the `Tagged` one, but that would require some changes in the Atlas metric infra (and for some reason in Tritium we don't have a `MetricRegistry asGenericRegistry()` or something like that on the tagged registry).